chore: Removed Menu in Preview (#2070)
Workday/canvas-kit: modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx

> **Note**: If you are consuming `StackProps` for a custom component, please make sure to change the
> `spacing` prop to the `gap` prop

---

> ### Menu in preview-react

Reviewer: Add this as a menu item up above so that we can link to it

Author: I think this got removed when dealing with merge conflicts. I'll re-add

> We've removed the `Menu` component that was located in `preview-react` and now will be replaced with

Reviewer: I think we reference thins in preview as `Preview` or `@workday/canvas-kit-preview-react`

Author: Updated

> the `Menu` component in `main`. Please see the `Menu` documentation

Reviewer: yeah im not too sure here, if we want to say `main` or just Main. with no back tickets, because we don't have a package named `main` it's `@workday/canvas-kit-react`

Reviewer: I'll defer to @jamesfan

          <DeprecatedMenuItem>
            Fourth Item (<b>with markup</b>)
          </DeprecatedMenuItem>
          <DeprecatedMenuItem hasDivider>Fifth Item (with divider)</DeprecatedMenuItem>
        </DeprecatedMenu>
      </Popup.Popper>
    </Popup>
  );
};
```

Using the `Menu` component from `main`:

```tsx
import React from 'react';
import {Menu} from '@workday/canvas-kit-react/menu';
import {BodyText} from '@workday/canvas-kit-react/text';

export const Basic = () => {
  const [selected, setSelected] = React.useState('');
  return (
    <Menu onSelect={data => setSelected(data.id)}>
      <Menu.Target>Open Menu</Menu.Target>
      <Menu.Popper>
        <Menu.Card>
          <Menu.List>
            <Menu.Item>First Item</Menu.Item>
            <Menu.Item>Second Item</Menu.Item>
            <Menu.Divider />
            <Menu.Item>Third Item (with a really, really, really long label)</Menu.Item>
            <Menu.Item>Fourth Item</Menu.Item>
          </Menu.List>
        </Menu.Card>
      </Menu.Popper>
      <BodyText size="small" marginTop="s">
        Selected: <span data-testid="output">{selected}</span>
      </BodyText>
    </Menu>
  );

> };

Reviewer: Curious if a full working example is needed or something like:

```
// Menu in @workday/canvas-kit-preview-react
<Popup model={model}>
      <Popup.Target
        as={SecondaryButton}
        onKeyDown={handleButtonKeyDown}
        aria-expanded={isOpen}
        aria-haspopup={true}
        aria-controls={isOpen ? menuId : undefined}
      >
        Open Menu
      </Popup.Target>
      <Popup.Popper placement="bottom-start">
        {/*
          isOpen must be set for focus to be properly assigned to the DeprecatedMenu;
          onClose must be set in order to the DeprecatedMenu to close properly after
          selecting a DeprecatedMenuItem
        */}
        <DeprecatedMenu id={menuId} isOpen={isOpen} onClose={model.events.hide}>
          <DeprecatedMenuItem>First Item</DeprecatedMenuItem>
          <DeprecatedMenuItem>
            Second Item (with a really really really long label)
          </DeprecatedMenuItem>
          <DeprecatedMenuItem isDisabled>Third Item (disabled)</DeprecatedMenuItem>
          <DeprecatedMenuItem>
            Fourth Item (<b>with markup</b>)
          </DeprecatedMenuItem>
          <DeprecatedMenuItem hasDivider>Fifth Item (with divider)</DeprecatedMenuItem>
        </DeprecatedMenu>
      </Popup.Popper>
    </Popup>
    
    // Menu from @workday/canvas-kit-react
    <Menu onSelect={data => setSelected(data.id)}>
      <Menu.Target>Open Menu</Menu.Target>
      <Menu.Popper>
        <Menu.Card>
          <Menu.List>
            <Menu.Item>First Item</Menu.Item>
            <Menu.Item>Second Item</Menu.Item>
            <Menu.Divider />
            <Menu.Item>Third Item (with a really, really, really long label)</Menu.Item>
            <Menu.Item>Fourth Item</Menu.Item>
          </Menu.List>
        </Menu.Card>
      </Menu.Popper>
      <BodyText size="small" marginTop="s">
        Selected: <span data-testid="output">{selected}</span>
      </BodyText>
    </Menu>
```

```

> **Note**: There will be no codemod to replace this component due to varying differences in API and

> > `Menu` is a compound component.

Reviewer: ```suggestion
> `Menu` being a compound component.
```

> **Note**: If you are consuming `StackProps` for a custom component, please make sure to change the
> `spacing` prop to the `gap` prop

---

### Menu in Preview

We've removed the `Menu` component that was in `@workday/canvas-kit-preview-react` and now will be
replaced with the `Menu` component in `@workday/canvas-kit-react`. Please see the `Menu`
documentation
[here](https://workday.github.io/canvas-kit/?path=/docs/components-popups-menu--basic).

For the migration guide from `Menu` in `@workday/canvas-kit-preview-react` to the `Menu` component

> in `@workday/canvas-kit-react`, see [this](https://github.com/Workday/canvas-kit/discussions/2063)

Reviewer: we can probably say `discussion` instead of `this`

discussion.

If you would like to see the differences between `DeprecatedMenu` and the `Menu` compound component,

> see [this](https://codesandbox.io/s/deprecatedmenu-to-menu-migration-gwi1ov) codesandbox example.

Reviewer: ```suggestion
see [this code sandbox](https://codesandbox.io/s/deprecatedmenu-to-menu-migration-gwi1ov) codesandbox example.
```

> ### Menu in Preview
> 
> We've removed the `Menu` component that was in `@workday/canvas-kit-preview-react` and now will be
> replaced with the `Menu` component in `@workday/canvas-kit-react`. Please see the `Menu`
> documentation
> [here](https://workday.github.io/canvas-kit/?path=/docs/components-popups-menu--basic).
> 
> For the migration guide from `Menu` in `@workday/canvas-kit-preview-react` to the `Menu` component
> in `@workday/canvas-kit-react`, see [this](https://github.com/Workday/canvas-kit/discussions/2063)
> discussion.
> 
> If you would like to see the differences between `DeprecatedMenu` and the `Menu` compound component,
> see [this](https://codesandbox.io/s/deprecatedmenu-to-menu-migration-gwi1ov) codesandbox example.
> 
> > **Note**: There will be no codemod to replace this component due to varying differences in API and
> > `Menu` being a compound component.

Reviewer 2: It was easier to have one big suggestion instead of several smaller ones. A few notes:

- To @mannycarrera4's point [here](https://github.com/Workday/canvas-kit/pull/2070#discussion_r1130071586) and [here](https://github.com/Workday/canvas-kit/pull/2070#discussion_r1130073797): we typically refer to the packages as "Preview" and "Main" (without backticks) for brevity. Sometimes, we'll write "the Preview package" or "the Main package." It is nice being explicit about the package names, though -- maybe we could add a short section at the top of the upgrade guide which defines Main, Preview, and Labs.
- When linking to other docs, you'll want to use the routes defined in [routes.js](https://github.com/Workday/canvas-kit/blob/master/.storybook/routes.js). For example, to link to Menu, use `/components/popups/menu/` instead of `https://workday.github.io/canvas-kit/?path=/docs/components-popups-menu--basic`. `/components/popups/menu/` will work correctly on the Canvas Site, whereas the `workday.github.io` link will take someone on the site to the Menu docs in Storybook instead of the Menu docs on the site.
- No need to include the codesandbox link here since it's already included in the linked migration discussion. **However**, I would consider moving the contents of that discussion into the upgrade guide itself. I took a quick look at the discussion, and there's some cleanup to be done, but I think it could go directly into the upgrade guide (at which point the codesandbox link would be added back to the guide). What do you think?

```suggestion
### Menu (Preview)

We've removed the `Menu` component in Preview. We recommend using the
[Menu](/components/popups/menu/) in the Main package instead; it is a
[compound component](/getting-started/for-developers/resources/compound-components/) and offers a
more flexible API. Please refer to this
[discussion](https://github.com/Workday/canvas-kit/discussions/2063) for instructions on how to
migrate from the `Menu` in Preview to the `Menu` in Main.

This change is **not** handled by the codemod due to the differences in API between the `Menu` in
Preview and the `Menu` in Main.
```
